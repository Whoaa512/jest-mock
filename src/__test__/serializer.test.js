import bookshelf from 'bookshelf';
import faker from 'faker';
import knex, { client } from '../';

const db = knex({
  client: 'sqlite',
  connection: { filename: ':memory:' },
  useNullAsDefault: true,
});
const { Model } = bookshelf(db);

const User = Model.extend({
  tableName: 'user',
  hasTimestamps: true,
});

describe('serializer', () => {
  it('client', async () => {
    const id = 99;
    const table = 'table';
    const limit = 100;
    const name = 'justin';
    const value = 'caffee';
    const at = faker.date.future();

    client.mockClear();
    client.mockReturnValueOnce([]);
    client.mockReturnValueOnce([]);
    client.mockReturnValueOnce([]);
    client.mockReturnValueOnce([]);

    const builder = db(table)
      .where({ id, at })
      .whereNull('deleted_at')
      .whereNotNull('nickname')
      .limit(limit);

    await builder.select('*');
    await builder.insert({ name, value, at });
    await builder.update({ name, value, at });
    await builder.delete();

    expect(client).toMatchSnapshot();
  });

  it('bookshelf', async () => {
    const id = 99;
    const name = 'justin';
    const value = 'caffee';
    const at = faker.date.future();

    client.mockClear();
    client.mockReturnValueOnce([]);
    client.mockReturnValueOnce([]);
    client.mockReturnValueOnce([]);
    client.mockReturnValueOnce([]);

    await User.fetchAll();
    await new User({ at, name, value }).save();
    await new User({ id, at, name, value }).save();
    await new User({ id }).destroy();

    expect(client).toMatchSnapshot();
  });
});
