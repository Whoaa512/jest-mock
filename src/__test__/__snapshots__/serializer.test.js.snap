// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`serializer bookshelf 1`] = `
Array [
  "select user.* from user",
  "insert into user (at, created_at, name, updated_at, value) values (DATE, DATE, justin, DATE, caffee)",
  "update user set at = DATE, id = 99, name = justin, updated_at = DATE, value = caffee where id = 99",
  "delete from user where id = 99",
]
`;

exports[`serializer client 1`] = `
Array [
  "select * from table where id = 99 and at = DATE and deleted_at is null and nickname is not null limit 100",
  "insert into table (at, name, value) values (DATE, justin, caffee)",
  "update table set at = DATE, name = justin, value = caffee where id = 99 and at = DATE and deleted_at is null and nickname is not null",
  "delete from table where id = 99 and at = DATE and deleted_at is null and nickname is not null",
]
`;
